test/e2e/xfn_test.go

```diff
@@ -106,24 +106,31 @@ func TestXfnRunnerImagePull(t *testing.T) {
 						))),
 			).
 			WithSetup("CopyFnImageToRegistry", func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
-				nodes := &corev1.NodeList{}
-				if err := config.Client().Resources().List(ctx, nodes); err != nil {
-					t.Fatal("cannot list nodes", err)
-				}
-				if len(nodes.Items) == 0 {
-					t.Fatalf("no nodes in the cluster")
-				}
-				var addr string
-				for _, a := range nodes.Items[0].Status.Addresses {
-					if a.Type == corev1.NodeInternalIP {
-						addr = a.Address
-						break
+				var reg string
+				_, found := envfuncs.GetKindClusterFromContext(ctx, clusterName)
+				if found {
+					reg = "localhost:3000"
```

Comment:
A container registry with custom TLS certificate is only needed in this test, so not sure why we should make the setup more complex/demanding for all other cases. Should a test not know that it runs under a certain flavor of k8s?

Reply:
How would it run locally if we were to use a loadbalancer? I was only suggesting making more evident the fact that this is tightly coupled to https://github.com/crossplane/crossplane/pull/4324/files#diff-6f3277362696d981ac73d0862a396840b1e0186813c8fd318297e783ca45aba4R8, but there is no way to know that, and if tomorrow we forget we get a broken test that will be hard to understand the reason of.

Reply:
just saw the link, sorry, no I think metalLB would be out of scope and I'm not sure it would solve the issue for docker on mac

Reply:
> ? I was only suggesting making more evident the fact that this is tightly coupled to https://github.com/crossplane/crossplane/pull/4324/files#diff-6f3277362696d981ac73d0862a396840b1e0186813c8fd318297e783ca45aba4R8, but there is no way to know that, and if tomorrow we forget we get a broken test that will be hard to understand the reason of.

Instead of setting here `localhost:3000`, we could create a function that returns ingress endpoint for the given service. That function could detect that we run against a kind cluster, parse its config file and return the proper endpoint, in this case `localhost:3000`. If we are not using kind, we can then add some other resolving mechanism. Wdyt? I think this is going to make things evident.

Reply:
@phisco pushed the change described above. Please let me know what do you think.